components/formats-bsd/src/loci/formats/out/OMETiffWriter.java

```diff
@@ -106,13 +113,30 @@ public void close() throws IOException {
           populateImage(omeMeta, series);
         }
 
+        String companion = getCompanion();
+        String companionUUID = null;
+        if (null != companion) {
+          String companionXML = getOMEXML(companion);
+          PrintWriter out = new PrintWriter(companion, Constants.ENCODING);
+          out.println(XMLTools.indentXML(companionXML, true));
+          out.close();
+          companionUUID = "urn:uuid:" + getUUID(
+              new Location(companion).getName());
+        }
+
```

Comment:
In order to keep the readability of the existing function I would be tempted to break this out this logic into a separate writeCompanion method and have it as part of the check further down. Something similar to the below:

```
String xml = null;
String companion = getCompanion();
if (null != companion) {
    String companionUUID = writeCompanion(companion);
     xml = getBinaryOnlyOMEXML(f, companion, companionUUID);
} else {

..................
private String writeCompanion(String companion) {
    String companionXML = getOMEXML(companion);
    PrintWriter out = new PrintWriter(companion, Constants.ENCODING);
    out.println(XMLTools.indentXML(companionXML, true));
    out.close();
    return "urn:uuid:" + getUUID(new Location(companion).getName());
}
```